subsys/bluetooth/host/conn.c

```diff
@@ -411,6 +415,11 @@ int bt_conn_send_cb(struct bt_conn *conn, struct net_buf *buf,
 	LOG_DBG("conn handle %u buf len %u cb %p user_data %p", conn->handle, buf->len, cb,
 		user_data);
 
+	if (buf->user_data_size < CONFIG_BT_L2CAP_TX_USER_DATA_SIZE) {
+		LOG_ERR("not enough room in user_data");
+		return -EINVAL;
+	}
```

Comment:
Can this happen with the BUILD_ASSERT?

Should this perhaps be an ASSERT instead of a LOG_ERR?

Reply:
yes it can happen, and people routinely disable asserts to save on flash, hence the runtime check.
It can happen because the API allows anyone, not just the networking stack, to pass buffers that they are responsible for allocating. I would also have preferred not having this at runtime if possible